03. DAL/NewHabr.DAL/Repository/Impl/ArticleRepository.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using NewHabr.DAL.EF;
+using NewHabr.Domain.Models;
+
+namespace NewHabr.DAL.Repository.Impl;
+public class ArticleRepository : IArticleRepository
+{
+    private readonly ILogger<ArticleRepository> _logger;
+    private readonly ApplicationContext _context;
+
+    public ArticleRepository(ILogger<ArticleRepository> logger, ApplicationContext context)
+    {
+        _logger = logger;
+        _context = context;
+    }
+
+    public async Task<Guid> Create(Article data)
+    {
+        await _context.Articles.AddAsync(data);
+        await _context.SaveChangesAsync();
```

Comment:
такие сохранения не дают делать изменения в рамках транзакций
например надо сохранить цепочку связанных элементов в бд, 
создали первый, на создании второй случилась беда. а первый элемент уже в базе без связи с вторым элементом

предлагаю создать класс менеджер репозиториев
```suggestion
        class RepositoryManager
        {
            public IArticleRepository Article => _articleRepository;
            ..
            public IUserRepository User => _userRepository;
            public async Task SaveAsync()
        }
```

Reply:
Менеджеры нужны, тут согласен. Но сохранение необходимо производить в репозитории.

Reply:
тут нужно пояснить, не догоняю.